### ISSUE TITLE
Store notice banner blocks content on small screens

### ISSUE THREAD

**Issue by mikejolley** [woocommerce/woocommerce#11539]
Needs responsive styling @jameskoster

**Comment by jameskoster** [comment]
I like this being sticky as it's quite important, when in use. As an enhancement I reckon we should make it dismissible.

Thinking about it some more however, do you think we should leave this in core? Feels kinda arbitrary and superfluous - there are _loads_ of 'notice' plugins on .org that do exactly the same thing.

**Comment by mikejolley** [comment]
@jameskoster The issue is it covers the content. Narrow screen makes the notice longer, but the padding at the top does not change to reflect this.

Not sure - maybe @mattyza can look at status. I guess most people just maintenance mode the site.

Saying that, it is useful for notices such as discounts or staff on leave on jobmanager.com :p

**Comment by jameskoster** [comment]
Yeah I know what the issue is, making the notice dismissible would resolve that whilst still keeping the information highlighted.

I'm not convinced it's all that useful, or even sees that much use. It's not doing any harm in core, but it's not exactly a crucial feature either which is why I'd suggest stripping it out.

**Comment by yuk75** [comment]
This notice is VERY useful and it's even more useful to have it in the core. I can't think of many softwares that don't have some sort of notice or maintenance message available and this is really handy....it just needs to be updated to be responsive or dismissable. Taking it out is not going to solve the issue.

**Comment by jameskoster** [comment]
> I can't think of many softwares that don't have some sort of notice or maintenance message available

How about WordPress? :p 

On a serious note, would installing a tiny plugin for this be a big deal? I'm not 100% against leaving it in, it just doesn't feel like a feature that's important enough to be in core.

**Comment by yuk75** [comment]
Having it available is important. Wordpress is a blog, it's not too often that a blog needs a global message...but for stores it can be pretty important to have a storewide message in some situations. For example if you change your currency, if you change your prices drastically, if you do something else drastic the customer needs to know or if something is broken or not working. If stock is out of stock and on backorder or shipping was delayed or what ever it can help prevent hundreds of email questions and alert customers to issues or information they need right away as they browse the store.

I think the fact that people notice it proves it's important. Having it in the software provides a quick and easy way to set it up....if it weren't in the software I would not have thought to add a plugin and with so many plugins added already I'd have avoided it and just plastered my notice externally - social media, email, blog post....and this would not have solved my issue.

In my case my store currency went from AUD to USD to match my customers expectations. I had many complaints on the price being higher than expected and had to explain that it was in AUD (Despite having a currency conversion plugin which was supposed to tell them it was AUD and show them their own currency conversion - the plugin didn't work on checkout or say which currency was where and listed two prices, and this is true of many Woocommerce plugins they are a mixed bag of hit and misses and most are paid which is why I would not have sought out a plugin, I can trust a built in feature (I thought) to handle it). I had dozens of emails about how to sign in/register and it was a mess. So I changed my storewide notice to say "Due to confusion, all prices listed are now USD. "Approximate prices" are an estimate based on the exchange rate. For help signing in hover over the "my account" menu item." Unfortunately 90% of my customers come in via mobile and this blocked them from selecting a shipping method or placing an order so all those issues it solved it may as well not have haha. I just pissed everyone off even more.

tl;dr: It's important to have something quick and accessible built into the software that we know we can trust to work with our themes and other plugins. If we want more than the basic functionality we can look into plugins which are often paid or not the best sure, but having it in the main software is extremely helpful and solves any issues quickly.

**Comment by jameskoster** [comment]
> It's important to have something quick and accessible built into the software that we know we can trust to work with our themes and other plugins. 

If we decide to remove it, we'd release a plugin which adds the functionality back, so that shouldn't be a problem.

**Comment by mattyza** [comment]
@jameskoster @mikejolley - I'd keep the notice in core, keep it sticky (not dismissable), and adjust the content/font-size for the narrower screen width.

**Comment by jameskoster** [comment]
That's not really going to work. We don't control the content and we won't be able to make the font size small enough to account for long messages. If it's sticky, it needs to be dismissible as well. If it's not sticky it's going to take up way too much room when the message is long. 

I still vote for sticky + dismissible, like all those horrible cookie notices we're subjected to.

**Comment by jameskoster** [comment]
Spoke about this, we're going to go with non-sticky and dismissible.

**Comment by carasmo** [comment]
When will this non-sticky and dismissible update happen? Thanks!

**Comment by carasmo** [comment]
It says that it's here (see below) on line 497, but it's not in my version 2.6.8

https://github.com/woocommerce/woocommerce/blob/4711066490cbbb650899dc1edd4a2e1ebdd22294/includes/wc-template-functions.php

**Comment by mikejolley** [comment]
2.7

**Comment by carasmo** [comment]
Cool! I will assume, since it will be relative or static positioned, it will be hooked into the wp_header? 

Thanks for replying. I am unhooking this and adding my own and want a slightly future proof way of doing it.

**Comment by mikejolley** [comment]
I don't think it's output has changed, just the CSS and content within the banner.

**Comment by carasmo** [comment]
Okay, however if it's non sticky, then it will appear at the very bottom of the site since it's hooked now to the wp_footer and shown at the top via position fixed top: 0. So if it gets unstickified, then it will need to be moved to the header or no one will see it, essentially.

**Comment by jameskoster** [comment]
@carasmo it's displayed at the top of the page using `position: absolute;`.

**Comment by carasmo** [comment]
Ahh. Brain must have been off line. Thanks!
